### PR TITLE
Menu: Support number pad keyboard input

### PR DESCRIPTION
**Pull request by fnagel**
Fixes #15031

### CODE REVIEW

**Comment by fnagel** [comment]
@arschmitz FYI

File: ui/widgets/menu.js [hunk 1]
```diff
@@ -236,9 +236,12 @@ return $.widget( "ui.menu", {
 		default:
 			preventDefault = false;
 			prev = this.previousFilter || "";
-			character = String.fromCharCode( event.keyCode );
 			skip = false;
 
+			// Support number pad values
+			character = event.keyCode >= 96 && event.keyCode <= 105 ?
+				event.keyCode - 96 : String.fromCharCode( event.keyCode );
```
Comment (arschmitz) [inline ui/widgets/menu.js]: I don't think this really matters in this case but when `event.keyCode >= 96 && event.keyCode <= 105` we get a number here and when it's less we get a string. Do we want to force character to always be a string for consistency?

File: tests/unit/selectmenu/core.js [hunk 2]
```diff
@@ -341,23 +341,40 @@ $.each( [
...
+	assert.equal( $.trim( widget.text() ), "Slower" );
+	form[ 0 ].reset();
+	setTimeout( function() {
+		assert.equal( $.trim( widget.text() ), initialValue );
+		ready();
 	} );
+} );
+
+// http://bugs.jqueryui.com/ticket/15031
+QUnit.test( "#15031: Number pad input should change value", function( assert ) {
```
Comment (arschmitz) [inline tests/unit/selectmenu/core.js]: We don't refrence bugs in tests like this in general any more i don't think

File: ui/widgets/menu.js (hunk 1 above)
Comment (scottgonzalez) [inline ui/widgets/menu.js]: I would.

Comment (arschmitz) [inline ui/widgets/menu.js]: that was my thought

Comment (fnagel) [inline ui/widgets/menu.js]: So adding something like `( event.keyCode - 9 ).toString();` would be suitable?

Comment (scottgonzalez) [inline ui/widgets/menu.js]: Yup, that'd be fine.

File: tests/unit/selectmenu/core.js (hunk 2 above)
Comment (fnagel) [inline tests/unit/selectmenu/core.js]: I've seen both (the prefix in the test title and the comment before the actual test) in the current code base. Which one are you talking about?

File: ui/widgets/menu.js (hunk 1 above)
Comment (fnagel) [inline ui/widgets/menu.js]: Fixed.

File: tests/unit/selectmenu/core.js (hunk 2 above)
Comment (arschmitz) [inline tests/unit/selectmenu/core.js]: Both actually while yes there are references in the code base like this currently this is something i'm pretty sure we don't do any more ( reference issues directly in tests ). 
@scottgonzalez

Comment (scottgonzalez) [inline tests/unit/selectmenu/core.js]: Correct. We only reference functionality now.

Comment (fnagel) [inline tests/unit/selectmenu/core.js]: Sooooo... I remove both ticket number references we're good to go?

Comment (scottgonzalez) [inline tests/unit/selectmenu/core.js]: Yup. The rest looks good :-)

Comment (fnagel) [inline tests/unit/selectmenu/core.js]: Fixed.

**Comment by scottgonzalez** [comment]
Looks good.

**Pull request closed without merging**